t/jv-anyof.t

```diff
@@ -63,4 +67,15 @@ validate_ok(
   E('/b', '/anyOf/1 Missing property.'),
 );
 
+validate_ok 'hello', {type => ['integer', 'string'], enum => [123, 'HELLO']},
+  E('/', 'Expected integer - got string.'),
```

Comment:
This doesn't sounds right. We got a valid type, didn't we..?

Reply:
fixed, by splitting apart the anyOf logic into two subs, and the array-type-specific version will disregard mismatched type errors if there are more specific errors we can report instead.